Delete MWAA

ividito opened the pull request:
## Summary
Starts to address #315 

- Remove MWAA infrastructure
- Remove MWAA CI
- Rework Makefile and scripts to match new folder structure (no more copying `dags` folder)
- SM2A moved to top level of repo
- Initial documentation refactor in #363 

## TODO

- [x] ~~Workflows API was managed with MWAA terraform. Should we keep it? We could move it into the SM2A infra easily enough, not sure what's needed for the runtime.~~
- [x] ~~The SM2A README can be combined with the root README, but I haven't done this yet~~ #363 
- [x] Merge https://github.com/NASA-IMPACT/veda-deploy/pull/113 when we merge this in
- [x] Some of the diffs don't look right (AFAIK many of the sm2a infra files that got marked as deleted were just moved, probably not an issue but I will double-check that the end result is what we intended)

This also unblocks additional work on #351 , where we can significantly improve Docker build times.

ividito committed:
Initial prune - remove MWAA

ividito committed:
More MWAA removal

ividito committed:
Accidentally included terraform graphs

ividito committed:
newline

smohiudd commented:
I think we should fully deprecate workflows API and remove it. We're not promoting it anymore as a way to ingest data

smohiudd commented:
Should we merge https://github.com/NASA-IMPACT/veda-data-airflow/pull/272 and https://github.com/NASA-IMPACT/veda-data-airflow/pull/297 first before merging this PR to minimize conflicts?

ividito commented:
@smohiudd #272 should definitely go in first, #297 we can discuss at tag-up (I know we were unsure about it - I tested that the feature flag works on SIT, I think we can merge it and decide whether to enable it or not later on)

ividito committed:
Reorganize docs

ividito committed:
Update README.md

Co-authored-by: Jennifer Tran <12633533+botanical@users.noreply.github.com>

ividito committed:
Doc changes from review
- add tags we should be using
- direct readers to DAG run templates, as well as veda-data examples

ividito committed:
Don't delete dags folder

ividito committed:
Remove workflows

ividito commented:
We'll deploy this to SIT using `veda-deploy` to test the CI, if that works this should be good to merge

ividito committed:
Try ci change

ividito committed:
Fix lambda trigger name

botanical commented on README.md:
```diff
@@ -27,30 +23,52 @@ See [terraform-getting-started](https://developer.hashicorp.com/terraform/tutori
- 
-- ⚠️ You need to copy ./sm2a/sm2a-local-config/env_example to ./sm2a/sm2a-local-config/.env and update the values of AWS secrets.
+### Configuration
+- ⚠️ You need to copy ./sm2a/sm2a-local-config/env_example to ./sm2a/sm2a-local-config/.env
```
```suggestion
- ⚠️ You need to copy ./sm2a-local-config/env_example to ./sm2a-local-config/.env
```

botanical replied:
Wait maybe this change isn't necessary? I think I'm a little confused

anayeaye replied:
or is it `⚠️ You need to copy ./sm2a-local-config/env_example to .env` because terraform will be looking for env config in the main directory

ividito replied:
I got confused by this too and had to double-check. Jennifer had the right change - the SM2A local runtimes will load the `.env` from the `local-config` folder, this is not the same as the `.env` used to deploy the project

botanical commented on docs/contributing/add_a_general_dag.md:
```diff
@@ -20,11 +46,10 @@
-- `ETL` - for ETL tasks outside of the Dynamic Vendor ETL Pipeline
-- `QAQC` - for data quality checks
-- `Template` - for templates
+- `Operations` - DAGs that are used for operational purposed, not for ingesting new data (fr example, scheduling and restoring backups)
```
```suggestion
- `Operations` - DAGs that are used for operational purposeds, not for ingesting new data (for example, scheduling and restoring backups)
```

botanical reviewed (approved):
Just some small nits. Thanks for tackling such a large PR

ividito committed:
Update docs/contributing/add_a_general_dag.md

Co-authored-by: Jennifer Tran <12633533+botanical@users.noreply.github.com>

ividito committed:
Update README.md

Co-authored-by: Jennifer Tran <12633533+botanical@users.noreply.github.com>
